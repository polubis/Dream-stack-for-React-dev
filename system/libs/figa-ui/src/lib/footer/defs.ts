import type { ReactNode } from 'react';

interface FooterProps {
  className?: string;
  socials: ReactNode;
  blocks: ReactNode;
  logo: ReactNode;
}

export type { FooterProps };
