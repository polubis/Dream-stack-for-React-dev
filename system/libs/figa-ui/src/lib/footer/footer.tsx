import type { FooterProps } from './defs';

import c from 'classnames';

const Footer = ({ className, socials, blocks, logo }: FooterProps) => {
  return (
    <div className={c('footer', className)}>
      <div className="footer-container">
        <div className="footer-content">{blocks}</div>

        <div className="footer-bar">
          <div className="footer-bar-left">{socials}</div>
          <div className="footer-bar-right">{logo}</div>
        </div>
      </div>
    </div>
  );
};

export { Footer };
