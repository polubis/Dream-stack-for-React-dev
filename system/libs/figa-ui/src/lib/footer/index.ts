export * from './defs';
export * from './footer';
