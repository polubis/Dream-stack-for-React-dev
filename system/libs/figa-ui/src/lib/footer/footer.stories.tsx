import type { Story, Meta } from '@storybook/react';

import { Footer } from './footer';
import { Font } from '../font';

export default {
  component: Footer,
  title: 'Footer',
} as Meta;

const Template: Story = () => {
  return (
    <Footer
      blocks={
        <>
          <Font variant="h5">First</Font>
          <Font variant="h5">Middle</Font>
          <Font variant="h5">Last</Font>
        </>
      }
      socials={<Font variant="b2">I</Font>}
      logo={<Font variant="b2">Logo</Font>}
    />
  );
};

export const Default = Template.bind({});
Default.args = {};
