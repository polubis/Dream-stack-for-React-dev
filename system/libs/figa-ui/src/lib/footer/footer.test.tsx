import { render, screen } from '@testing-library/react';

import { Footer } from './footer';

describe('Footer can be used when', () => {
  const footerFixture = () => {
    const result = render(
      <Footer
        logo={<div>Logo</div>}
        socials={<div>I</div>}
        blocks={
          <>
            <div>A</div>
            <div>B</div>
            <div>C</div>
          </>
        }
      />
    );

    return result;
  };

  it('[FRAGILE] renders with classes', () => {
    const { asFragment } = footerFixture();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders blocks, socials and logo', () => {
    footerFixture();

    screen.getByText('A');
    screen.getByText('B');
    screen.getByText('C');
    screen.getByText('I');
    screen.getByText('Logo');
  });
});
