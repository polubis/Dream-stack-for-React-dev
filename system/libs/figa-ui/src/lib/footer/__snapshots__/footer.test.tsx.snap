// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer can be used when [FRAGILE] renders with classes 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <div
      class="footer-container"
    >
      <div
        class="footer-content"
      >
        <div>
          A
        </div>
        <div>
          B
        </div>
        <div>
          C
        </div>
      </div>
      <div
        class="footer-bar"
      >
        <div
          class="footer-bar-left"
        >
          <div>
            I
          </div>
        </div>
        <div
          class="footer-bar-right"
        >
          <div>
            Logo
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
